lib/tmuxinator/cli.rb

```diff
@@ -69,25 +71,86 @@ def completions(arg)
                           aliases: ["-l"],
                           desc: "Create local project file at ./.tmuxinator.yml"
 
-    def new(name)
-      project_file = find_project_file(name, options[:local])
-      Kernel.system("$EDITOR #{project_file}") || doctor
+    def new(name, session = nil)
+      if session
+        new_project_with_session(name, session)
+      else
+        new_project(name)
+      end
     end
 
     no_commands do
+      def new_project(name)
+        project_file = find_project_file(name, options[:local])
+        Kernel.system("$EDITOR #{project_file}") || doctor
+      end
+
+      def new_project_with_session(name, session)
```

Comment:
I had in mind breaking this into a few different, single-purpose methods, and testing each of them individually.  As you noted in another comment, this method is too large; I see [a version check](https://github.com/pczarn/tmuxinator/blob/12de4a1dc8e5b595c6d0bd1037298ac150ae9c23/lib/tmuxinator/cli.rb#L88-L91), [a set of shell interactions](https://github.com/pczarn/tmuxinator/blob/12de4a1dc8e5b595c6d0bd1037298ac150ae9c23/lib/tmuxinator/cli.rb#L93-L103), [a shell interaction check](https://github.com/pczarn/tmuxinator/blob/12de4a1dc8e5b595c6d0bd1037298ac150ae9c23/lib/tmuxinator/cli.rb#L105-L107), [a data structure creation](https://github.com/pczarn/tmuxinator/blob/12de4a1dc8e5b595c6d0bd1037298ac150ae9c23/lib/tmuxinator/cli.rb#L109-L120), [a data transformation](https://github.com/pczarn/tmuxinator/blob/12de4a1dc8e5b595c6d0bd1037298ac150ae9c23/lib/tmuxinator/cli.rb#L122-L133) and finally [an I/O operation](https://github.com/pczarn/tmuxinator/blob/12de4a1dc8e5b595c6d0bd1037298ac150ae9c23/lib/tmuxinator/cli.rb#L135-L138).  I'm not saying we need to break this into 6 separate methods, but some breakdown along those lines, which enables some input/output unit tests, is probably a good idea.

If you're asking me...  I think the version check could be useful in other places.  The shell interactions can be difficult to unit test, so I'd pull those out into their own method(s), probably wrapped up in a facade method that includes the success check.  I'd likely combine the data structure creation/transformation into it's own method, which should be easily unit-tested (e.g. `build_yaml_from(panes, windows: [])`), and put the I/O operation into it's own helper method (if there isn't one already... I would guess that there is one, but I could be wrong).  For what it's worth, I'd recommend putting the bulk of this functionality into a separate module, which should be easily testable standalone (the CLI is kinda a pain to test directly).